rest_framework/permissions.py

```diff
@@ -112,7 +112,8 @@ def has_permission(self, request, view):
         except AttributeError:
             queryset = getattr(view, 'queryset', None)
         except AssertionError:
```

Comment:
We def don't need the `AssertionError` catching in either case, but it might also be nicer to move `if getattr(view, '_ignore_model_permissions', False): return True` behavior to instead be *above* the queryset check.